internal/datastore/proxy/context.go

```diff
@@ -0,0 +1,119 @@
+package proxy
```

Comment:
Do we want this in the proxy package or did we want to make a new middleware package?

Reply:
Open question; I just put it there for the draft